File: apprefurbished/src/main/java/com/androidcollider/easyfin/faq/RecyclerFAQAdapter.java
package com.androidcollider.easyfin.faq;

import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.util.Pair;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.androidcollider.easyfin.R;

import java.util.List;

import static butterknife.ButterKnife.findById;

/**
 * @author Ihor Bilous
 */

class RecyclerFAQAdapter extends RecyclerView.Adapter<RecyclerFAQAdapter.ViewHolder> {

    private List<Pair<String, String>> itemsList;


    RecyclerFAQAdapter(List<Pair<String, String>> itemsList) {
        this.itemsList = itemsList;
    }

    @Override
    public int getItemCount() {
        return itemsList.size();
    }

    @Override
    public RecyclerFAQAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new RecyclerFAQAdapter.ViewHolder(LayoutInflater.from(parent.getContext()).inflate(R.layout.item_faq, parent, false));
    }

    private Pair<String, String> getItem(int position) {
        return itemsList.get(position);
    }

    @Override
    public void onBindViewHolder(final RecyclerFAQAdapter.ViewHolder holder, final int position) {
        Pair<String, String> pair = getItem(position);
        holder.tvHead.setText(pair.first);
        holder.tvBody.setText(pair.second);
        holder.card.setOnClickListener(v ->
                holder.tvBody.setVisibility(
                        holder.tvBody.getVisibility() == View.GONE ?
                                View.VISIBLE : View.GONE)
        );
    }


    static class ViewHolder extends RecyclerView.ViewHolder {
        private final CardView card;
        private final TextView tvHead;
        private final TextView tvBody;


        ViewHolder(View view) {
            super(view);
            card = findById(view, R.id.cardFAQ);
            tvHead = findById(view, R.id.tvFAQHead);
            tvBody = findById(view, R.id.tvFAQBody);
        }
    }
}

File: apprefurbished/src/main/java/com/androidcollider/easyfin/faq/RecyclerFAQAdapter.kt
package com.androidcollider.easyfin.faq

import android.util.Pair
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import com.androidcollider.easyfin.R
import com.androidcollider.easyfin.common.utils.setSafeOnClickListener

/**
 * @author Ihor Bilous
 */
internal class RecyclerFAQAdapter : RecyclerView.Adapter<RecyclerFAQAdapter.ViewHolder>() {

    private val items: MutableList<Pair<String, String>> = ArrayList()

    fun setData(itemList: List<Pair<String, String>>) {
        items.clear()
        items.addAll(itemList)
        notifyItemRangeInserted(0, itemCount)
    }

    override fun getItemCount(): Int {
        return items.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
            LayoutInflater.from(parent.context)
                .inflate(R.layout.item_faq, parent, false)
        )
    }

    private fun getItem(position: Int): Pair<String, String> {
        return items[position]
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val pair = getItem(position)
        holder.tvHead.text = pair.first
        holder.tvBody.text = pair.second
        holder.card.setSafeOnClickListener {
            holder.tvBody.visibility =
                if (holder.tvBody.visibility == View.GONE) View.VISIBLE else View.GONE
        }
    }

    internal class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        val card: CardView
        val tvHead: TextView
        val tvBody: TextView

        init {
            card = view.findViewById(R.id.cardFAQ)
            tvHead = view.findViewById(R.id.tvFAQHead)
            tvBody = view.findViewById(R.id.tvFAQBody)
        }
    }
}